full-service/src/db/transaction_log.rs

```diff
@@ -777,16 +777,16 @@ impl TransactionLogModel for TransactionLog {
     }
 
     fn update_pending_associated_with_txo_to_succeeded(
-        txo_id_hex: &str,
+        txo_out_id_hex: &str,
         finalized_block_index: u64,
         conn: Conn,
     ) -> Result<(), WalletDbError> {
-        use crate::db::schema::{transaction_input_txos, transaction_logs};
+        use crate::db::schema::{transaction_logs, transaction_output_txos};
         // Find all submitted transaction logs associated with this txo that have not
         // yet been finalized (there should only ever be one).
         let transaction_log_ids: Vec<String> = transaction_logs::table
-            .inner_join(transaction_input_txos::table)
-            .filter(transaction_input_txos::txo_id.eq(txo_id_hex))
+            .inner_join(transaction_output_txos::table)
+            .filter(transaction_output_txos::txo_id.eq(txo_out_id_hex))
             .filter(transaction_logs::submitted_block_index.is_not_null()) // we actually sent this transaction
```

Comment:
I think making the minimal change is the correct, conservative approach.

The finalized_block_index check might also be unnecessary at this point.

However, if we go stripping those things out and it turns out that there are ways to get multiple transaction_log entries that have a common output txo, this may prevent issues for any "shadow" log entries that are not the actual submitted transaction.

There should only be 1 transaction_log entry that matches, and so perhaps a check for that and some sort of log message would be the change for the next PR on the topic?